src/composables/queries/usePoolsQuery.ts

```diff
@@ -77,14 +80,16 @@ export default function usePoolsQuery(
         const poolDecorator = new PoolDecorator(pools);
         let decoratedPools = await poolDecorator.decorate(tokenMeta.value);
 
-        const tokens = flatten(
-          pools.map(pool => [
-            ...pool.tokensList,
-            ...tokenTreeLeafs(pool.tokens),
-            pool.address,
-          ])
-        );
-        await injectTokens(tokens);
+        if (shouldInjectTokens) {
+          const tokens = flatten(
+            pools.map(pool => [
+              ...pool.tokensList,
+              ...tokenTreeLeafs(pool.tokens),
+              pool.address,
+            ])
+          );
+          await injectTokens(tokens);
```

Comment:
shouldn't `await` be removed?

Reply:
We don't really use the subgraph case anymore and we'll be completely removing it soon. But I think when we fetch pools from the subgraph we do have a need to await the injection. Could be wrong though. Don't think it really matters right now.